src/devtools/client/debugger/src/components/Editor/NewSourceAdapter.tsx

```diff
@@ -65,11 +66,12 @@ function NewSourceAdapter() {
     // Sync focused state from Redux to React context,
     if (
       focusedSource?.sourceId !== location.sourceId ||
-      focusedSource?.startLineIndex !== lineIndex
+      focusedSource?.startLineIndex !== lineIndex ||
+      !locationHasScrolled
```

Comment:
Even if the location hasn't changed, this flag indicates that we need to re-open and re-scroll the source back into view because the user has re-clicked on it.